File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

sys.setrecursionlimit(10**5)

def dfs(v):
     print(v, end=' ')
     visited[v] = True
     for e in adj[v]:
          if not(visited[e]):
               dfs(e)
def bfs(v):
     q = deque([v])
     while q:
          v= q.popleft()
          if not(visited[v]):
               visited[v] = True
               print(v, end=' ')
               for e in adj[v]:
                    if not visited[e]:
                         q.append(e)

v, e ,start = map(int,sys.stdin.readline().split())
adj = [[]for i in range(v+1)]
for i in range(e):
    a, b = map(int,sys.stdin.readline().split())
    adj[a].append(b)
    adj[b].append(a)

for i in adj:
    i.sort()

visited = [False]*(v+1)
dfs(start)
print()
visited = [False]*(v+1)
bfs(start)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <bits/stdc++.h>
using namespace std;

vector<int> adj[10002];
vector<int> result_bfs;
vector<int> result_dfs;
bool visited[1002];

void dfs(int x)
{
	visited[x] = true;
	result_dfs.push_back(x);
	for (int i = 0; i < adj[x].size(); ++i)
	{
		if (!visited[adj[x][i]])
			dfs(adj[x][i]);
	}
}

void bfs(int x)
{
	queue<int> q;
	q.push(x);
	visited[x] = true;
	while (!q.empty())
	{
		int tmp = q.front();
		q.pop();
		result_bfs.push_back(tmp);
		for (int i = 0; i < adj[tmp].size(); ++i)
		{
			if (!visited[adj[tmp][i]])
			{
				q.push(adj[tmp][i]);
				visited[adj[tmp][i]] = true;
			}
		}
	}
}

int main()
{
	ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
	int node, edge, start;
	cin >> node >> edge >> start;
	for (int i = 0; i < edge; ++i)
	{
		int a, b;
		cin >> a >> b;
		adj[a].push_back(b);
		adj[b].push_back(a);
	}
	for (int i = 1; i <= node; ++i)
	{
		sort(adj[i].begin(), adj[i].end());
	}
	bfs(start);
	memset(visited, false, sizeof(visited));
	dfs(start);
	for (auto i : result_dfs)
		cout << i << " ";
	cout << "\n";
	for (auto i : result_bfs)
		cout << i << " ";
	
	return 0;
}
